src/platform/backends/hyperv_api/hcs/hyperv_hcs_wrapper.cpp

```diff
@@ -0,0 +1,474 @@
+               compute_system_name,
+               file_path.string());
+
+    const auto path_as_wstring = file_path.wstring();
+    const std::wstring csname_as_wstring = maybe_widen{compute_system_name};
+    const auto result = API().HcsRevokeVmAccess(csname_as_wstring.c_str(), path_as_wstring.c_str());
+    return {result, FAILED(result) ? L"RevokeVmAccess failed!" : L""};
+}
+
+// ---------------------------------------------------------
+
+OperationResult HCSWrapper::get_compute_system_state(const HcsSystemHandle& target_hcs_system,
+                                                     ComputeSystemState& state_out) const
+{
+    mpl::debug(log_category,
+               "get_compute_system_state(...) > handle: ({})",
+               fmt::ptr(target_hcs_system.get()));
+
+    const auto result = perform_hcs_operation(
+        [&](auto&& op) {
+            return API().HcsGetComputeSystemProperties(
+                static_cast<HCS_SYSTEM>(target_hcs_system.get()),
+                op,
+                nullptr);
+        },
+        target_hcs_system);
+
+    if (!result)
+        return result;
+
+    state_out = [json = result.status_msg]() {
+        QString qstr{QString::fromStdWString(json)};
+        const auto doc = QJsonDocument::fromJson(qstr.toUtf8());
+        const auto obj = doc.object();
+        if (obj.contains("State"))
+        {
+            const auto state = obj["State"];
+            const auto state_str = state.toString();
+            const auto ccs = compute_system_state_from_string(state_str.toStdString());
+            if (ccs)
+            {
+                return ccs.value();
+            }
+            return ComputeSystemState::unknown;
+        }
+        return ComputeSystemState::stopped;
+    }();
+
+    return {result.code, L""};
+}
+
+// ---------------------------------------------------------
+
+OperationResult HCSWrapper::modify_compute_system(const HcsSystemHandle& target_hcs_system,
+                                                  const HcsRequest& params) const
+{
+    mpl::debug(log_category,
+               "modify_compute_system(...) > handle: ({}), params: {}",
+               fmt::ptr(target_hcs_system.get()),
+               params);
+
+    const auto json = fmt::to_wstring(params);
+
+    return perform_hcs_operation(
+        [&](auto&& op) {
+            return API().HcsModifyComputeSystem(static_cast<HCS_SYSTEM>(target_hcs_system.get()),
+                                                op,
+                                                json.c_str(),
+                                                nullptr);
+        },
+        target_hcs_system);
+}
+
+// ---------------------------------------------------------
+
+OperationResult HCSWrapper::set_compute_system_callback(const HcsSystemHandle& target_hcs_system,
+                                                        void* context,
+                                                        void (*callback)(void* hcs_event,
```

Comment:
Should it avoid the `void*` for the event here too (and call chains this is involved in)?